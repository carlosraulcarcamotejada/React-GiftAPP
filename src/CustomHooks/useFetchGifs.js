import { useState, useEffect } from 'react';
import getGifs from '../utilities/getGifs';

const useFetchGifs = (category) => {

    const [data, setData] = useState(
        {
            data: [],
            loading: true
        }
    )

    useEffect(() => {
        getGifs(category)
            .then(
                (imgs) => {
                    setData({
                        data: imgs,
                        loading: false
                    });
                   
                }
            );

    }, [category]);


    return data;
}


export default useFetchGifs;