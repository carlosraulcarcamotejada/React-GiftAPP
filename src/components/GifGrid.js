import React from 'react';
import useFetchGifs from '../CustomHooks/useFetchGifs';
import { GifItem } from './GifItem';
import PropTypes from 'prop-types'


export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);
    

    return (
        <>
            <h3> {category}  </h3>
            { loading && <p >Loading...</p>}
            {
                images.map(image =>
                    <GifItem
                        key={image.id}
                        {...image}
                    />)
            }

        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}