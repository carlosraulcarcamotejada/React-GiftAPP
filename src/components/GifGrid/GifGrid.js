import React from 'react';
import useFetchGifs from '../../CustomHooks/useFetchGifs';
import GifItem from '../GifItem/GifItem';

const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3> {category} </h3>

            {
                images.map(image =>
                    <GifItem
                        key={image.id}
                        {...image}
                    />)
            }

        </>
    )
}


export default GifGrid;