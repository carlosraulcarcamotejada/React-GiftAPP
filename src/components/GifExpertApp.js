import React, { useState } from 'react'
import { AddCategory } from './AddCategory'
import { GifGrid } from './GifGrid'




export const GifExpertApp = ({ categoriesDefault = [] }) => {

    const [categories, setCategories] = useState(categoriesDefault)



    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory
                setCategories={setCategories}
            />
            <hr />

            {
                categories.map(category => {
                    return (<GifGrid key={category} category={category} />);
                })
            }

        </>
    )
}

