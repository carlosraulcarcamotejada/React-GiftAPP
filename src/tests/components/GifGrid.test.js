import { shallow } from 'enzyme'
import React from 'react'
import { GifGrid } from '../../components/GifGrid';
import useFetchGifs from '../../CustomHooks/useFetchGifs';
jest.mock('../../CustomHooks/useFetchGifs');
import '@testing-library/jest-dom';

describe('Pruebas sobre el componente <GifGrid />', () => {

    const category = 'onn punch';


    test('Test 1. ', () => {


        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Test 2. debe de mostrar items cuando se cargan imagenes useFetch', () => {

        const gifs = [{
            id: '0001',
            url: 'https://localhost/data/cualquier-cosa.jpg',
            title: 'cualquier cosa'
        }];


        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifItem').length).toBe(gifs.length);
    })

 



})