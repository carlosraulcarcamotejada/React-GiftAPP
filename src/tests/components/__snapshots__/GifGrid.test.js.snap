// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <GifGrid /> Test 1.  1`] = `
<Fragment>
  <h3>
     
    onn punch
      
  </h3>
  <p>
    Loading...
  </p>
</Fragment>
`;

exports[`Pruebas sobre el componente <GifGrid /> Test 2. debe de mostrar items cuando se cargan imagenes useFetch 1`] = `
<Fragment>
  <h3>
     
    onn punch
      
  </h3>
  <GifItem
    id="0001"
    key="0001"
    title="cualquier cosa"
    url="https://localhost/data/cualquier-cosa.jpg"
  />
</Fragment>
`;
